docs/compiling.rst

```diff
@@ -181,9 +183,8 @@ to an independently constructed (through ``add_library``, not
     flags (i.e. this is up to you).
 
     These include Link Time Optimization (``-flto`` on GCC/Clang/ICPC, ``/GL``
-    and ``/LTCG`` on Visual Studio). Default-hidden symbols on GCC/Clang/ICPC
-    (``-fvisibility=hidden``) and .OBJ files with many sections on Visual Studio
-    (``/bigobj``). The :ref:`FAQ <faq:symhidden>` contains an
+    and ``/LTCG`` on Visual Studio) and .OBJ files with many sections on Visual
+    Studio (``/bigobj``). The :ref:`FAQ <faq:symhidden>` contains an
```

Comment:
The condition should be as simple as `if(NOT MSVC)`, shouldn't it?

After some thinking, I feel that the hidden visibility should be enforced as much as possible. With an increasing number of pybind11 modules on PyPI, it would be quite easy to run into `internals` versioning issues and/or bindings of common STL containers.